- Documentation for Indi component (#143)
elasticpath/react-pwa-reference-storefront: documentation/practices.md

For any other CMS, you must update the configurations to reflect the public URLs of the files being retrieved by the particular content provider.

## Indi Brand Loyalty Example

The React PWA Reference Storefront is pre-configured for integration with Indi, with a custom component created to interact with the various widgets Indi may display. For more information on Indi, visit their official [website](https://indi.com/).

The Indi component available is a stand-alone component created solely to fetch the `indi-embed` libraries, then display the corresponding elements that were passed into the component as a structured list.

The required parameters of this component are as follows:

> * `render`: A list of the elements to display from Indi. Example: `render={['carousel', 'brand', 'product']}`. Note the correct configurations must be provided for the element to render correctly.

Reviewer: I'm just looking at the comments, but one question I have as someone new (and maybe if I saw the whole page, this would make sense) but in the note, you mention correct configurations are required, is there an example of some configurations?  If I were to do it, I would want a guide.

> * `render`: A list of the elements to display from Indi. Example: `render={['carousel', 'brand', 'product']}`. Note the correct configurations must be provided for the element to render correctly.

Author: Example provided in `ep.config.json`

* `configuration`: The structured object consisting of the configurations to provide to the Indi components. These are defined in `ep.config.json` in the `indi` element.

> * `keywords`: A string of the keywords for the product in which the Indi component is displayed for.

Author: @sminnath It's the former - `for the product for which the indi component is displayed`

I'll update for that 👍

> Static strings for the Indi component have been localized in the Reference Storefront and are overrided on a per-usage basis for the Indi component. This will ensure the component displays the correct text in the storefront based on the locale which a customer chooses during their shopping flow.

Reviewer 2: ```suggestion
Static strings for the Indi component are localized in the Reference Storefront and are overridden whenever the Indi component is used. With this setting, the component displays the storefront in the locale that a customer chooses.
```

Reviewer 2: Why is this section here? Why are we suddenly talking about Static strings here when in the next section we are going to explain `configuration`?

Author: I wanted to call out that we provide localization options for the strings a developer would normally pass to the Indi library to display the component. This allows for other languages to be used for the component dynamically rather than hardcoding the strings through configurations.

Reviewer 2: Do you want to have a heading here? @shaunmaharaj

> 

Reviewer 2: Are you specifically talking about the configuration in the  `ep.config.json` in the `indi` parameter? 
If yes this should be a sub section with the title 
**Configuration Settings for Indi in the ep.config.json file**

Author: Nope, I'm referring to the files in `src/localization`.
Both of these two:
`src/localization/en-CA.json`
`src/localization/fr-FR.json`

> Configuration properties for Indi component's requirements are:

Reviewer 2: Following are the configuration settings for the `indi` parameter in the `ep.config.json` file:

> * `indi.enable`: Enable the integration component for Indi. More information may be found available here [https://indi.com/](https://indi.com/).

Reviewer 2: ```suggestion
* `indi.enable`: The parameter that enables the component for the integration with Indi. For more information about the required configurations, see the company [website](https://indi.com/).
```

> * `indi.carousel`: Configurations for the Indi carousel component.

Reviewer 2: The configurations for the Indi carousel component.

> * `indi.carousel.apikey`: The apikey used to connect the carousel to Indi.

Reviewer 2: ```suggestion
* `indi.carousel.apikey`: The _apikey_ that connects a carousel to Indi.
```
use the carousel if it is for the above mentioned one

> * `indi.carousel.id`: The apikey used to connect the carousel to Indi.

Reviewer 2: this doesnt look correct. You mean id?

> * `indi.carousel.size`: Used when theming the carousel displaying content from Indi.

Reviewer 2: ```suggestion
* `indi.carousel.size`: The parameter used to theme the carousel that displays the content from Indi.
```

* `indi.carousel.round_corners`: Used when theming the carousel displaying content from Indi.
* `indi.carousel.show_title`: Used when theming the carousel displaying content from Indi.
* `indi.carousel.show_views`: Used when theming the carousel displaying content from Indi.
* `indi.carousel.show_likes`: Used when theming the carousel displaying content from Indi.
* `indi.carousel.show_buzz`: Used when theming the carousel displaying content from Indi.
* `indi.carousel.animate`: Used when theming the carousel displaying content from Indi.
* `indi.productReview`: Configurations for the Indi product review component.
* `indi.productReview.submit_button_url`: The URL provided by Indi which directs to the action performed when the submit button is clicked on a product review.
* `indi.productReview.thumbnail_url`: The thumbnail provided by Indi for the submit button.
* `indi.brandAmbassador`: Configurations for the Indi brand ambassador component.
* `indi.brandAmbassador.submit_button_url`: The URL provided by Indi which directs to the action performed when the submit button is clicked on a brand ambassador signup.

> * `indi.brandAmbassador.thumbnail_url`: The thumbnail provided by Indi for the submit button.

Reviewer 2: Can we have a table with the following details:

Functionality|     Parameter |            Description |  >> **Table header**

**Integration**

**Theming**          |  List all parameters from (101 to 108)|    Description for 101
                                                                                                    ...

**Product View** .  |  (line 109 - 111)|

**Brand ambassador** |(line 112-114) |

Author: Adding in latest commit

For any other CMS, you must update the configurations to reflect the public URLs of the files being retrieved by the particular content provider.

##  Third-Party Dynamic Content Integration Example

**Brand Loyalty Integration for Indi**

> The React PWA Reference Storefront is pre-configured with a custom component for Indi integration. This component interacts with various widgets that Indi displays. New widgets may be added in the component as made available. For more information on Indi, see the company [website](https://indi.com/).

Reviewer 2: When Indi adds new widgets, you can update the component with the new widget.

The Indi component is a stand-alone component to fetch only the `indi-embed` libraries and display the elements corresponding to the libraries. These elements are passed to the component as a structured list.

This component requires the following parameters:

> * `render`: The list of the elements to display from Indi. For example, `render={['carousel', 'brand', 'product']}`.

Reviewer 2: Is there a specific format for this parameter? or should we just say. 
For example, see the `ep.config.json' file settings, `render={['carousel', 'brand', 'product']}`.

Author: Yep there's a format. I'll need to keep the example

> Note:  Provide the correct configurations to render the element  correctly.

Reviewer 2: extra space after element

> * `configuration`: The structured object that consists of the configuration required for the Indi components. These configurations are defined in the `indi` parameter in `react-pwa-reference-storefront⁩/ep.config.json' file.

Reviewer 2: ```suggestion
* `configuration`: The structured object that consists of the configuration required for the Indi components. These configurations are defined in the `indi` parameter in the `react-pwa-reference-storefront⁩/ep.config.json' file.
```

* `keywords`: A string of the keywords for the product in which the Indi component is displayed.

> Static strings for the Indi component are localized in the Reference Storefront and are overridden whenever the Indi component is used. With this setting, the component displays the storefront in the locale that a customer chooses. The localized strings for the Indi component can be found in the corresponding language files within `src/localization`, for strings with the prefix `indi-`.

Reviewer 2: ```suggestion
Static strings for the Indi component are localized in the Reference Storefront and are overridden whenever the component is used. With this setting, the component displays the storefront in the locale that a customer chooses. The localized strings for the Indi component are available in the corresponding language files in the `src/localization` directory. This setting is only for the strings with the `indi-`prefix.
```

| **Integration**|`indi.enable`| The parameter that enables the component for the integration with Indi. For more information about the required configurations, see the company [website](https://indi.com/).|
| **Theming**|`indi.carousel`| The configurations for the Indi carousel component.|
| |`indi.carousel.apikey`| The _apikey_ that connects a carousel to Indi.|
| |`indi.carousel.id`| The id used to connect the carousel to Indi.|
| |`indi.carousel.size`| The parameter used to theme the carousel that displays the content from Indi.|
| |`indi.carousel.theme`| The parameter used to theme the carousel that displays the content from Indi.|
| |`indi.carousel.round_corners`| The parameter used to theme the carousel that displays the content from Indi.|
| |`indi.carousel.show_title`| The parameter used to theme the carousel that displays the content from Indi.|
| |`indi.carousel.show_views`| The parameter used to theme the carousel that displays the content from Indi.|
| |`indi.carousel.show_likes`| The parameter used to theme the carousel that displays the content from Indi.|
| |`indi.carousel.show_buzz`| The parameter used to theme the carousel that displays the content from Indi.|

> | |`indi.carousel.animate`| The parameter used to theme the carousel that displays the content from Indi.|

Reviewer 2: Can we merge all rows with one `The parameter used to theme the carousel that displays the content from Indi.`

> | **Product Review**|`indi.productReview`| Configurations for the Indi product review component.|

Reviewer 2: The configurations

> | |`indi.productReview.submit_button_url`| The URL provided by Indi which directs to the action performed when the submit button is clicked on a product review.|

Reviewer 2: ```suggestion
| |`indi.productReview.submit_button_url`| The URL of the page to which Indi redirects when a brand ambassador submits a product review.

| |`indi.productReview.thumbnail_url`| The thumbnail provided by Indi for the submit button.|

> | **Brand Ambassador**|`indi.brandAmbassador`| Configurations for the Indi brand ambassador component.|

Reviewer 2: ```suggestion
| **Brand Ambassador**|`indi.brandAmbassador`| The configurations for the Indi brand ambassador component.|
```

Reviewer 2: Is the component name brand ambassador?

Author: The config for it is

Reviewer 2: configuration for brand ambassador component in Indi, right?

Author: yes, that's correct

> | |`indi.brandAmbassador.submit_button_url`| The URL provided by Indi which directs to the action performed when the submit button is clicked on a brand ambassador signup.|

Reviewer 2: ```suggestion
| |`indi.brandAmbassador.submit_button_url`| The URL to which Indi redirects when a brand ambassador signs up.|
```
